test/src/test/java/org/ethereum/beacon/test/runner/state/StateRunner.java

```diff
@@ -1,103 +1,83 @@
-    try {
-      spec = buildSpec(testCase);
-    } catch (Exception e) {
-      return Optional.of("Failed to build BeaconChainSpec: " + e.getMessage());
-    }
-
-    BeaconStateEx initialState = buildInitialState(spec, testCase.getInitialState());
-    Optional<String> err = StateComparator.compare(testCase.getInitialState(), initialState);
+    BeaconState initialState = buildInitialState(spec, testCase.getPre());
+    Optional<String> err = StateComparator.compare(testCase.getPre(), initialState);
     if (err.isPresent()) {
       return Optional.of("Initial state parsed incorrectly: " + err.get());
     }
+    BeaconState latestState = initialState;
 
-    EmptySlotTransition preBlockTransition = StateTransitions.preBlockTransition(spec);
-    BlockTransition<BeaconStateEx> blockTransition = StateTransitions.blockTransition(spec);
-
-    BeaconStateEx latestState = initialState;
-    for (StateTestCase.BlockData blockData : testCase.getBlocks()) {
-      Pair<BeaconBlock, Optional<String>> blockPair = StateTestUtils.parseBlockData(blockData);
-      if (blockPair.getValue1().isPresent()) {
-        return blockPair.getValue1();
-      }
-      BeaconBlock block = blockPair.getValue0();
-
-      BeaconStateEx postBlockState;
-      try {
-        BeaconStateEx preBlockState = preBlockTransition.apply(latestState, block.getSlot());
-        postBlockState = blockTransition.apply(preBlockState, block);
-      } catch (Exception ex) {
-        StringWriter sw = new StringWriter();
-        PrintWriter pw = new PrintWriter(sw);
-        ex.printStackTrace(pw);
-        return Optional.of("Error happened during transition: " + sw.toString());
-      }
-      latestState = postBlockState;
+    if (testCase.getDeposit() != null) {
+      processDeposit(testCase.getDepositOperation(), latestState);
+    } else if (testCase.getAttestation() != null) {
+      processAttestation(testCase.getAttestationOperation(), latestState);
+    } else {
+      throw new RuntimeException("Only Attestation and Deposit test cases are implemented!!!");
     }
 
-    return StateComparator.compare(testCase.getExpectedState(), latestState);
+    if (testCase.getPost() == null) { // XXX: Not changed
+      return StateComparator.compare(testCase.getPre(), latestState);
+    } else {
+      return StateComparator.compare(testCase.getPost(), latestState);
+    }
   }
 
-  private BeaconStateEx buildInitialState(BeaconChainSpec spec, BeaconStateData stateData) {
-    BeaconState state = StateTestUtils.parseBeaconState(spec.getConstants(), stateData);
-    return new BeaconStateExImpl(state, TransitionType.BLOCK);
+  private void processDeposit(Deposit deposit, BeaconState state) {
+    try {
+      DepositVerifier depositVerifier = new DepositVerifier(spec);
+      assert depositVerifier.verify(deposit, state).isPassed();
+      spec.process_deposit((MutableBeaconState) state, deposit);
+    } catch (SpecCommons.SpecAssertionFailed | AssertionError ex) {
```

Comment:
We don't use `assert` in the spec, do we? I am thinking of some rare cases when `AssertionError` has been due to odd state of some object (usual way `assert` is made for); in that case I'd rather prefer to see this exception than get it swallowed here.